File: src/main/java/no/nav/syfo/repository/CarDAO.java
package no.nav.syfo.repository;

import no.nav.syfo.domain.Car;
import org.springframework.jdbc.core.RowMapper;
import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
import org.springframework.stereotype.Repository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import javax.inject.Inject;
import java.util.List;
import java.util.UUID;

@Service
@Transactional
@Repository
public class CarDAO {

    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;

    public CarDAO(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
        this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
    }

    public String lagreCar(final Car car) {
        String uuid = UUID.randomUUID().toString();

        namedParameterJdbcTemplate.update(
                "INSERT INTO CAR VALUES(:uuid, :color, :brand, :bought)",
                new MapSqlParameterSource()
                        .addValue("uuid", uuid)
                        .addValue("color", car.getColor())
                        .addValue("brand", car.getBrand())
                        .addValue("bought", car.getBought())
        );

        return uuid;
    }

    public List<Car> hentCars() {
        return namedParameterJdbcTemplate.query("SELECT * FROM CAR", getInnsendingRowMapper());
    }

    public static RowMapper<Car> getInnsendingRowMapper() {
        return (resultSet, i) -> Car.builder()
                .uuid(resultSet.getString("CAR_UUID"))
                .color(resultSet.getString("COLOR"))
                .brand(resultSet.getString("BRAND"))
                .bought(resultSet.getDate("BOUGHT").toLocalDate())
                .build();
    }
}

File: src/main/kotlin/no/nav/syfo/repository/CarDAO.kt
package no.nav.syfo.repository

import no.nav.syfo.domain.Car
import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
import org.springframework.stereotype.Repository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.sql.ResultSet
import java.util.*

@Service
@Transactional
@Repository
class CarDAO(private val namedParameterJdbcTemplate: NamedParameterJdbcTemplate) {

    val innsendingRowMapper: (ResultSet, Int) -> Car = { resultSet, _ ->
        Car(
            uuid = (resultSet.getString("CAR_UUID")),
            color = (resultSet.getString("COLOR")),
            brand = (resultSet.getString("BRAND")),
            bought = (resultSet.getDate("BOUGHT").toLocalDate())
        )
    }

    fun lagreCar(car: Car): String {
        val uuid = UUID.randomUUID().toString()

        namedParameterJdbcTemplate.update(
            "INSERT INTO CAR VALUES(:uuid, :color, :brand, :bought)",
            MapSqlParameterSource()
                .addValue("uuid", uuid)
                .addValue("color", car.color)
                .addValue("brand", car.brand)
                .addValue("bought", car.bought)
        )

        return uuid
    }

    fun hentCars(): List<Car> {
        return namedParameterJdbcTemplate.query("SELECT * FROM CAR", innsendingRowMapper)
    }
}
